app/rest/rest/src/main/java/io/syndesis/rest/v1/handler/metrics/IntegrationMetricsHandler.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.rest.v1.handler.metrics;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.springframework.stereotype.Component;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.syndesis.dao.manager.DataManager;
+import io.syndesis.model.Kind;
+import io.syndesis.model.ListResult;
+import io.syndesis.model.metrics.MetricsSummary;
+import io.syndesis.rest.v1.handler.BaseHandler;
+import io.syndesis.rest.v1.operations.Getter;
+
+@Path("/metrics/integrations")
+@Api(value = "integration-metrics")
+@Component
+public class IntegrationMetricsHandler extends BaseHandler
```

Comment:
This shouldn't extend `BaseHandler` as it ties the source of metrics to jsondb via the `DataManager`. Please separate this out and create an interface that can be implemented differently depending on the respective backend source. Then plug in the `DataManager` into your jsondb implementation. I can then work on the Prometheus implementation.